furbaby/api/views.py

```diff
@@ -177,3 +187,125 @@ def password_reset_token_created(sender, instance, reset_password_token, *args,
+        request.data["user_id"] = request.user.id
+        return location_view.insert_location_record(request.data)
+
+    # update a location record for the user
+    if request.method in ["PUT", "PATCH"]:
+        return location_view.update_location_record(request.user.id, request.data)
+
+    # delete a location record for the user
+    if request.method == "DELETE":
+        return location_view.delete_location_record(request.user.id, request.data)
+
+    return json_response(
+        {"error": "incorrect request method supplied"},
+        status=status.HTTP_405_METHOD_NOT_ALLOWED,
+    )
+
+
+# This class is for the user location(s)
+class UserLocationView(APIView):
+    authentication_classes = []
+
+    # Fetch the locations serializer
+    serializer_class = UserLocationSerializer
+
+    def get_exception_handler(self):
+        return exception_handler
+
+    # takes as input the user id, request and inserts a new location record for the user
+    def insert_location_record(self, request_data):
+        serializer = self.serializer_class(data=request_data)
+
+        if not serializer.is_valid():
+            return json_response(data=serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        instance = serializer.save()
+
+        return json_response(
+            instance.id, status=status.HTTP_201_CREATED, include_data=False, safe=False
+        )
+
+    # takes as input a location object and returns a dictionary of the location key value pairs
+    def get_location_record(self, location=None):
+        if location == None:
+            return None
+        return {
+            "id": location.id,
+            "address": location.address,
+            "city": location.city,
+            "country": location.country,
+            "zipcode": location.zipcode,
+            "user_id": location.user_id,
+            "default_location": location.default_location,
+        }
+
+    # takes as input a user_id and returns a JSON of all the locations for that user
+    def get_user_locations(self, user_id):
+        locations = Locations.objects.filter(user_id=user_id)
+        serialized_data = serialize("json", locations)
+        serialized_data = json.loads(serialized_data)
+        return serialized_data
+
+    # takes as input a location_id and location fields and updates the location record
+    def update_location_record(self, user_id, request_data):
+        try:
+            location_id = request_data["id"]
+            location = Locations.objects.get(id=location_id)
+            location.address = request_data["address"]
+            location.city = request_data["city"]
+            location.country = request_data["country"]
+            location.zipcode = request_data["zipcode"]
+            location.default_location = request_data["default_location"]
+            location.save()
+            return json_response(
+                self.get_location_record(Locations.objects.get(id=location_id)),
```

Comment:
you could just return the updated `location` object directly, the `get` is quite redundant here.